client/my-sites/email/add-mailboxes/index.tsx

```diff
@@ -300,7 +308,7 @@ const MailboxesForm = ( {
 		cartManager
 			.addProductsToCart( [ getCartItems( mailboxOperations.mailboxes, mailProperties ) ] )
 			.then( () => {
-				page( '/checkout/' + selectedSite.slug );
+				page( '/checkout/' + selectedSite?.slug ?? '' );
```

Comment:
This will prevent redirecting to the invalid URL `/checkout/undefined` and will use the valid URL `/checkout/` instead.